contracts/pcv/ERC4626/ERC4626PCVDeposit.sol

```diff
@@ -0,0 +1,283 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity =0.8.13;
+
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {SafeCast} from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import {CoreRefV2} from "../../refs/CoreRefV2.sol";
+import {IPCVOracle} from "../../oracle/IPCVOracle.sol";
+import {PCVDeposit} from "../PCVDeposit.sol";
+import {IERC4626} from "./IERC4626.sol";
+
+/// @notice Generic PCV Deposit for a ERC4626 "Tokenized Vault"
+/// - Implements the PCV Deposit interface to deposit and withdraw funds to/from a vault
+/// - Implements a specific function "withdrawMax", slightly changing from "withdrawAll"
+///   function that can be seen in other PCV Deposit. The reason is that the ERC4626 Vault
+///   can restrict the amount of withdrawable tokens but still report the total token held
+/// - Implements withdrawableBalance() which allow to get the amount of withdrawable tokens
+/// @dev The underlying token in the Deposit is named "token" but is named "asset" in the Vault.
+/// @dev The ERC4626 gives back "shares" when an asset is deposited. The shares are only used
+/// in the vault internal accounting. The PCV Deposit does not show any values in "shares"
+/// but always in "asset" (token) as explicitly stated in the function "balanceReportedIn()"
+/// It should be noted that the balance() function use the 'share' amount to compute the
+/// amount of token currently in possession of the PCV Deposit (with profit & loss)
+contract ERC4626PCVDeposit is PCVDeposit {
+    using SafeERC20 for IERC20;
+    using SafeCast for *;
+
+    /// ------------------------------------------
+    /// ---------- Immutables/Constant -----------
+    /// ------------------------------------------
+
+    /// @notice reference to ERC4626 vault
+    address public immutable vault;
+
+    /// @notice reference to underlying token
+    /// @dev private because the accessor would be a duplicate
+    ///      of the function 'balanceReportedIn'
+    address private immutable token;
+
+    /// ------------------------------------------
+    /// ------------- State Variables -------------
+    /// ------------------------------------------
+
+    /// @notice track the last amount of PCV recorded in the contract
+    /// this is the total token deposited - total token withdrawn + total PNL
+    /// this is always out of date, except when accrue() is called
+    /// in the same block or transaction. This means the value is stale
+    /// most of the time.
+    uint256 public lastRecordedBalance;
+
+    /// @param _core reference to the core contract
+    /// @param _underlying Token denomination of this deposit
+    /// @param _vault the reference to the ERC4626 Vault the PCVDeposit interfaces
+    constructor(
+        address _core,
+        address _underlying,
+        address _vault
```

Comment:
Would be nice to add a constructor parameter `bool _isLiquid` (and associated public state variable) so that the PCVDeposit knows if it should report itself as a liquid balance or illiquid balance to the PCV Oracle